src/instructlab/sdg/utils/docprocessor.py

```diff
@@ -0,0 +1,595 @@
+            max_token_per_chunk=500,
+            tokenizer=self.tokenizer,
+        )
+        chunks = fuse_texts(chunks, 200)
+        return Dataset.from_dict(
+            {
+                "document": chunks,
+                "document_outline": [self.qna_yaml.get("document_outline", "")]
+                * len(chunks),
+                "document_title": [file_name] * len(chunks),
+                "domain": [self.qna_yaml.get("domain", "")] * len(chunks),
+            }
+        )
+
+    def _add_icls(self, chunked_document: Dataset) -> Dataset:
+        """
+        Add the ICLS label to the dataset.
+        Args:
+            chunked_document (Dataset): Dataset object.
+        Returns:
+            Dataset: Dataset object with ICLS label.
+        """
+        icl = self.qna_yaml.get("seed_examples", [])
+        chunked_document_all_icl = []
+        for icl_ in icl:
+            chunked_document_all_icl.append(
+                chunked_document.map(
+                    lambda x: {
+                        "icl_document": icl_["context"],
+                        "icl_query_1": icl_["questions_and_answers"][0]["question"],
+                        "icl_response_1": icl_["questions_and_answers"][0]["answer"],
+                        "icl_query_2": icl_["questions_and_answers"][1]["question"],
+                        "icl_response_2": icl_["questions_and_answers"][1]["answer"],
+                        "icl_query_3": icl_["questions_and_answers"][2]["question"],
+                        "icl_response_3": icl_["questions_and_answers"][2]["answer"],
+                    }
+                )
+            )
+        chunked_document_all_icl = safe_concatenate_datasets(chunked_document_all_icl)
+        chunked_document_all_icl = chunked_document_all_icl.map(
+            lambda x: {
+                "chunks": chunk_markdowns(
+                    [x["document"]], server_ctx_size=4096, chunk_word_count=1024
+                )
+                if get_token_count(x["document"], self.tokenizer) > 1024
+                else [x["document"]]
+            }
+        )
+        df = chunked_document_all_icl.to_pandas()
+        df_exploded = df.explode("chunks").reset_index(drop=True)
+        new_ds = Dataset.from_pandas(df_exploded)
+        new_ds = new_ds.remove_columns("document").rename_columns(
+            {"chunks": "document"}
+        )
+
+        # Only keep document greater than 100 tokens
+        new_ds = new_ds.filter(
+            lambda x: get_token_count(x["document"], self.tokenizer) > 100
+        )
+        return new_ds
+
+    def get_processed_dataset(self) -> Dataset:
+        """
+        Process all the parsed docling json files and return a dataset.
+        Returns:
+            Dataset: Dataset object.
+        """
+        datasets = []
+        for json_fp in self.docling_jsons:
+            chunk_ds = self._process_parsed_docling_json(json_fp)
+            chunk_ds_with_icls = self._add_icls(chunk_ds)
```

Comment:
Where do we add the icl values for non-PDF files? This codepath only happens for PDFs, right?

Reply:
Ok, digging into this a bit more, we shouldn't be doing anything different for icl values with PDFs, right? I'm confused why we do anything with icl_query_* here at all, as that all gets handled in taxonomy.py already. What's the intent behind the separate duplicate icl handling for pdfs?